### PR TITLE
#163519136 Add email notification on user signup

### PR DESCRIPTION
**Pull request by oesukam**
#### What does this PR do?
Adds email notification on user signup

#### Description of Task to be completed?
- Install @sengrid/mail package
- Create a mail controller
- Add test for mail controller
- Add mail notification after user signed up
- Implement email confirmation

#### How should this be manually tested?
- git clone this branch

- run `npm install`

- run `npm migrate:seed`

- run `npm run dev`

- with postman use this link to signup:
`POST: api/v1/users/`

- Body request example (login):
```request-json
{
   "user": {
       "username": "username",
       "email": "username@gmail.com"
       "password": "123456"
   }
}
```
- At last check your mail inbox for the confirmation email

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
`#163519136`

#### Screenshots (if appropriate)
- Confirmation Email Link sent to oemapp.dev@gmail.com
image

- Email confirmed sent to oemapp.dev@gmail.com
image

#### Questions:
N/A

### CODE REVIEW

**Comment by kimotho-njoki** [comment]
@oesukam tag reviewers on this PR. Two atleast.

**Comment by oesukam** [comment]
@mcaleb808 @filschristian Please notify if you have already reviewed this PR.

**Comment by oesukam** [comment]
@mcaleb808 @filschristian Can you please review this PR and notify when it is done.

**Review by filschristian — APPROVED** [review]
It is okay every thing is running fine... it can be merged if we discover what's wrong with travis

File: controllers/MailController.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import 'dotenv/config';
+import { sendgrid } from '../helpers';
+
+const { FRONTEND_URL = '/' } = process.env;
+
+/**
+ * @author Olivier
+ * @param {Object} user
+ * @returns {Promise} - Returns a promise
```
Comment (kimotho-njoki) [inline controllers/MailController.js:10]: `@returns {Promise}` gives enough context. No need for a comment beside the statement.

File: controllers/MailController.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import 'dotenv/config';
```
[the remark this answers is not in the document]
Comment (kimotho-njoki) [reply controllers/MailController.js]: Why are we ignoring this hound errors?

**Comment by oesukam** [comment]
@kimotho-njoki The PR is ready to be merged

**Review by mcaleb808 — APPROVED** [review]
I reviewed the PR, I approve merging

File: controllers/MailController.js (hunk 1 above)
Comment (oesukam) [reply controllers/MailController.js:10]: @kimotho-njoki eslint complains if I don't provide a description following @returns {Promise}

File: .env.example [hunk 3]
```diff
@@ -1,2 +1,4 @@
 DATABASE_URL=postgres://username:password@localhost:5432/db_name
 JWT_SECRET=secret
+SENDGRID_API_KEY=
+FRONTEND_URL=http://localhost:3000
```
Comment (kimotho-njoki) [inline .env.example]: Can we also add an extra line after line 4

**Comment by oesukam** [comment]
@filschristian @mcaleb808 I have completed the last subtask. You can do the last review

File: .env.example (hunk 3 above)
Comment (oesukam) [reply .env.example]: @kimotho-njoki Yes, I have just added the extra line

File: controllers/MailController.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (oesukam) [reply controllers/MailController.js]: I have already fixed the issue

**Pull request merged**